fs/fs.go

```diff
@@ -30,12 +31,22 @@ func CreateHomeConfigFolder(folder string) string {
 	}
 	return path
 }
+
+// CreateSecureFolder checks if the folder exists and has the appropriate permission rights. If not it creates it.
 func CreateSecureFolder(folder string) string {
 	if exists, _ := Exists(folder); !exists {
 		if err := os.MkdirAll(folder, 0740); err != nil {
 			fmt.Println("folder", folder, ",err", err)
 			panic(err)
 		}
+	} else {
+		//the folder exists already
+		info, err := os.Lstat(folder)
+		perm := int(info.Mode().Perm())
+		//given that the permission is incremental (owner then group then others), then can compare the value of the permission bytes
+		if err != nil || perm < int(0740) {
```

Comment:
I am not sure I understand the reasoning. If the folder has perm `0777` then it is still considered as a "secure folder" ?
I think we should aim for an equality here, i.e. ` perm == 0740` no ?